src/docs/release/breaking-changes/route-transition-record-and-transition-delegate.md

```diff
@@ -0,0 +1,139 @@
+---
+title: The Route Transition Record and Transition Delegate Updates
+description: Changes to the rule on how transition delegate resolve route transition.
+---
+
+## Summary
+
+Added a new boolean getter `isWaitingForExitingDecision` to the route
+transition record and renamed the getter `isEntering` to
+`isWaitingForEnteringDecision`. During the resolve method in transition
+delegate, developers need to use the `isWaitingForExitingDecision` to check
+if a exiting route actually needs an explicit decision on how to transition
+off the screen. Otherwise, it throws assertion errors if developers try to
+make decisions for the exiting routes that are not waiting for exiting
+decisions.
+
+
+## Context
+
+If the navigator receives a new list of pages, it tries to update its
+current routes stack to match the list. It, however, requires explicit
+decisions on how to transition the route in and off the screen. Developers
+can customize that by providing their own implementations of transition
```

Comment:
=> You can customize its behavior by providing your own implementation for the transition delegate.